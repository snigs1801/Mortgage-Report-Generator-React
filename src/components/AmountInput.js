import React from "react";

const AmountInput = ({ labelName, targetName, targetValue, setMortgageAmount }) => {

  const regex = /^[0-9]*$/;
  
  // mortgage amount input component
  return (
    <div className="input-group">
      <label>{labelName}</label>
      <div className="input-box">
        <div className="add-on right-align">CAD</div>
        <input
          type="text"
          name={targetName}
          className="left-align"
          value={targetValue}
          placeholder="Mortgae Amount"
          onChange={e => (e.target.value !== '' && regex.test(e.target.value)) ? setMortgageAmount(e.target.value) : setMortgageAmount('') }
        />
      </div>
    </div>
  );
};

export default AmountInput;