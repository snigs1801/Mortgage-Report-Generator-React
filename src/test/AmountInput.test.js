import React from 'react';import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AmountInput from "../components/AmountInput"; 

Enzyme.configure({ adapter: new Adapter() });

describe('AmountInput component Test Cases', () => {
  const container = shallow(<AmountInput />);
  
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should have a input text field', () => { 
    expect(container.find('input[type="text"]').length).toEqual(1);
  });
});
