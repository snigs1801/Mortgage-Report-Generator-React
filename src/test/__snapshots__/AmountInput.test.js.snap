// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountInput component Test Cases should match the snapshot 1`] = `"<div class=\\"input-group\\"><label></label><div class=\\"input-box\\"><div class=\\"add-on right-align\\">CAD</div><input type=\\"text\\" class=\\"left-align\\" placeholder=\\"Mortgae Amount\\"/></div></div>"`;
